avocado/plugins/runners/vmimage.py

```diff
@@ -0,0 +1,114 @@
+import sys
+import traceback
+from multiprocessing import set_start_method
+
+from avocado.core.nrunner.app import BaseRunnerApp
+from avocado.core.nrunner.runner import BaseRunner
+from avocado.core.utils import messages
+from avocado.plugins.vmimage import download_image
+from avocado.utils import vmimage
+
+
+class VMImageRunner(BaseRunner):
+    """
+    Runner for dependencies of type vmimage.
+    This runner uses the vmimage plugin's download_image function which handles:
+    1. Checking if the image exists in cache
+    2. Downloading the image if not in cache
+    3. Storing the image in the configured cache directory
+    4. Returning the cached image path
+    """
+
+    name = "vmimage"
+    description = "Runner for dependencies of type vmimage"
+
+    def run(self, runnable):
+        try:
+            yield messages.StartedMessage.get()
```

Comment:
While the image is being downloaded the runner should send running messages to avocado about the status. Therefore, this should run in subprocess. See [avocado-instrumented](https://github.com/avocado-framework/avocado/blob/master/avocado/plugins/runners/avocado_instrumented.py#L155)